helix-core/src/test/java/org/apache/helix/task/TestTargetedTaskStateChange.java

```diff
@@ -0,0 +1,330 @@
+    currentStateOutput.setResourceStateModelDef(JOB_NAME, "TASK");
+    currentStateOutput.setBucketSize(JOB_NAME, 0);
+    Partition taskPartition = new Partition(JOB_NAME + "_" + PARTITION_NAME);
+    currentStateOutput.setEndTime(JOB_NAME, taskPartition, masterInstance, 0L);
+    currentStateOutput.setCurrentState(JOB_NAME, taskPartition, masterInstance, masterState);
+    currentStateOutput.setInfo(JOB_NAME, taskPartition, masterInstance, "");
+    currentStateOutput.setResourceStateModelDef(TARGET_RESOURCES, "MasterSlave");
+    currentStateOutput.setBucketSize(TARGET_RESOURCES, 0);
+    Partition dbPartition = new Partition(TARGET_RESOURCES + "_0");
+    currentStateOutput.setEndTime(TARGET_RESOURCES, dbPartition, masterInstance, 0L);
+    currentStateOutput.setCurrentState(TARGET_RESOURCES, dbPartition, masterInstance, "MASTER");
+    currentStateOutput.setInfo(TARGET_RESOURCES, dbPartition, masterInstance, "");
+    return currentStateOutput;
+  }
+
+  private ResourceAssignment preparePreviousAssignment(String instance, String state) {
+    ResourceAssignment prevAssignment = new ResourceAssignment(JOB_NAME);
+    Map<String, String> replicaMap = new HashMap<>();
+    replicaMap.put(instance, state);
+    Partition taskPartition = new Partition(JOB_NAME + "_" + PARTITION_NAME);
+    prevAssignment.addReplicaMap(taskPartition, replicaMap);
+    return prevAssignment;
+  }
+
+  private class MockTestInformation {
+    private String slaveInstance = INSTANCE_PREFIX + "0";
+    private String masterInstance = INSTANCE_PREFIX + "1";
+    private String slaveInstance2 = INSTANCE_PREFIX + "2";
+    private WorkflowControllerDataProvider cache = mock(WorkflowControllerDataProvider.class);
+    private WorkflowConfig _workflowConfig = prepareWorkflowConfig();
+    private WorkflowContext _workflowContext = prepareWorkflowContext();
+    private Map<String, IdealState> _idealStates =
+        prepareIdealStates(masterInstance, slaveInstance, slaveInstance2);
+    private JobConfig _jobConfig = prepareJobConfig();
+    private JobContext _jobContext = prepareJobContext(slaveInstance);
+    private BestPossibleStateOutput _bestPossibleStateOutput = mock(BestPossibleStateOutput.class);
+    private CurrentStateOutput _currentStateOutput = prepareCurrentState(masterInstance,
+        slaveInstance, TaskPartitionState.RUNNING.name(), TaskPartitionState.RUNNING.name());
+    private CurrentStateOutput _currentStateOutput2 =
+        prepareCurrentState2(masterInstance, TaskPartitionState.RUNNING.name());
+    private ResourceAssignment _resourceAssignment =
+        preparePreviousAssignment(slaveInstance, TaskPartitionState.RUNNING.name());
+    private ResourceAssignment _resourceAssignment2 =
+        preparePreviousAssignment(slaveInstance, TaskPartitionState.DROPPED.name());
+    private TaskDataCache _taskDataCache = mock(TaskDataCache.class);
+    private RuntimeJobDag _runtimeJobDag = mock(RuntimeJobDag.class);
```

Comment:
Let's be consistent with underscores here as well? some have them, others don't.

Reply:
Good point. Fixed.